### PR TITLE
feat: upgrade to new version of pinecone with serverless api

### PR DESCRIPTION
**Pull request by MikaAK**
Brings changes inline with new pinecone api so we can access serverless indexes

### CODE REVIEW

File: lib/pinecone.ex [hunk 1]
```diff
@@ -105,40 +108,72 @@ defmodule Pinecone do
...
+        validate!("replicas", pod_opts[:replicas], :non_negative_integer)
+        validate!("shards", pod_opts[:shards], :non_negative_integer)
+        validate!("pod_type", pod_opts[:pod_type], :one_of, @valid_pods)
+
+        %{"pod" => %{
+          "pods" => pod_opts[:pods],
+          "replicas" => pod_opts[:replicas],
+          "shards" => pod_opts[:shards],
+          "pod_type" => to_pod_type(pod_opts[:pod_type])
+        }}
```
Comment (seanmor5) [inline lib/pinecone.ex:173]: I think you want to run the formatter here :)

Comment (MikaAK) [reply lib/pinecone.ex:173]: Ah yes, I'm notoriously bad with running formatter 😂  I'll do that

**Comment by MikaAK** [comment]
Hey @seanmor5, happy to take it over for you 😄  I'm using it actively in a startup I'm running so I'm more than happy to maintain the library and keep it updated. 

Happy to create a test setup for this all though to keep it in good condition, I can give it the standard library package treatment I do with my other libs!

**Comment by seanmor5** [comment]
@MikaAK Sounds good to me :) I will transfer it to you now! Thanks for taking it on from me! I'm glad somebody will be able to keep it up to date

**Comment by seanmor5** [comment]
Ah so apparently I cannot transfer it to you because you already have a forked version, so instead I've just invited you to the repo and I will make you a repo owner. If you have another org or something you'd like it transferred to in the future I'm happy to do that as well

**Comment by MikaAK** [comment]
Oh, strange, I'll get this passing and we can merge it in, then we can do the transfer after I delete my existing fork 😵‍💫 I'll try to have it all wrapped up by EoD

**Comment by MikaAK** [comment]
@seanmor5  I've removed my fork on github 😄

**Pull request merged**